javascript/javascript1/week4/Voice_assistant.js

```diff
@@ -0,0 +1,101 @@
+let userName = "";
+let userAge = -1;
+let todos = [];
+
+function getReply(command){
+    if (command.startsWith("Hello my name is")) {
+        const name = command.slice(17).trim();
+        if (userName && userName.toLowerCase() === name.toLowerCase()) {
+            return `You already told me your name is ${userName}!`;
+        }
+        userName = name;
+        return `Nice to meet you ${userName}`;
+    }
+
+    if (command.startsWith("What is my name")) {
+        if (userName === "") {
+            return `Firstly add your name!`;
+        }
+        return `Your name is ${userName}`;
+    }
+
+    if (command.startsWith("How old I am")) {
+        if (userAge === -1) {
+            return `Firstly add your age!`;
+        }
+        return `Your age is ${userAge}`;
+    }
+
+    if (command.startsWith("My age is")) {
+        if (userAge === -1) {
+            const age = parseInt(command.slice(10).trim());
+            if(typeof(age) !== "number" || age < 0){
+                return "Incorrect age!"
+            }
+            userAge = age;
+            return `You added your age (${userAge} y.o.)`;
+        }
+        return `You have already added your age`;
+    }
+
+    if (command.startsWith("Add")) {
+        const task = command.slice(4, command.lastIndexOf("to my todo")).trim();
+        todos.push(task);
+        return `${task} added to your todo`;
+    }
+
+    if (command.startsWith("Remove")) {
+        const task = command.slice(7, command.lastIndexOf("from my todo")).trim();
+        const index = todos.indexOf(task);
+        if (index !== -1) {
+            todos.splice(index, 1);
+            return `Removed ${task} from your todo`;
+        }
+        return `${task} is not in your todo list`;
+    }
+
+    if (command.startsWith("What is on my todo")) {
+        if(todos.length < 1){
+            return "Todo list is empty";
+        }
+        return `Your todos:\n\n${todos.join(", \n")}.\n`;
+    }
+
+    if (command.startsWith("What day is it today")) {
+        const today = new Date();
+        const day = today.getDate();
+        const month = today.toLocaleString("default", { month: "long" });
+        const year = today.getFullYear();
+        return `Today is ${day}. of ${month} ${year}`;
+    }
+
+    if (command.startsWith("What is")) {
+        const mathExpression = command.slice(8).trim();
+        const result = eval(mathExpression);
```

Comment:
Replace eval with a safer alternative. For example, you can use the Function constructor in strict mode to evaluate expressions securely.